src/js/components/back-to-top/back-to-top.component.js

```diff
@@ -38,13 +48,21 @@ class BackToTop extends LitElement {
   }
 
   getNextScrollPosition() {
-    return Math.max(0, this.getScrollPosition() - 50);
+    return Math.max(this.getTopOfPage(), this.getScrollPosition() - 50);
   }
 
   getScrollPosition() {
     return window.pageYOffset || document.documentElement.scrollTop;
   }
 
+  getTopOfPage() {
+    if (this.target) {
+      const elem = document.getElementById(this.target);
+      return elem.getBoundingClientRect().top + document.documentElement.scrollTop
```

Comment:
Do you want to throw in an `if (elem === null) return 0;` statement?

Reply:
Good thinking. Added a check for the target element